pages/api/getNonce.ts

```diff
@@ -0,0 +1,37 @@
+import { createClient } from '@supabase/supabase-js'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { getSupabase } from '../../utils/supabase'
+
+type Data =
+  | {
+      nonce: number
+    }
+  | {
+      error: string
+    }
+
+const supabase = getSupabase(process.env.SUPABASE_SERVICE_ROLE as string)
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse<Data>) {
+  try {
+    const nonce = Math.floor(Math.random() * 10000)
+
+    const { walletAddress } = await JSON.parse(req.body)
+    const { data, error: getUserError } = await supabase.from('users').select('*').eq('address', walletAddress)
+    if (getUserError) throw new Error('getUserError: ' + getUserError.message)
+
+    if (data.length > 0) {
+      const { error: updateUserError } = await supabase.from('users').update({ nonce: nonce }).eq('id', data[0].id)
```

Comment:
Do we want to record only the latest nonce? Why not all of them?

Reply:
We just need the latest nonce to accurately verify the user is the one says to be

Reply:
For future clarity. We just need some place to save the nonce. For now it's in the DB.